crates/sui-indexer-alt-framework/src/lib.rs

```diff
@@ -1067,4 +1078,344 @@ mod tests {
         assert_eq!(watermark1.unwrap().checkpoint_hi_inclusive, 19);
         assert_eq!(watermark2.unwrap().checkpoint_hi_inclusive, 19);
     }
+
+    /// When a tasked indexer is initialized such that a tasked pipeline is run with a
+    /// `first_checkpoint` less than the main pipeline's reader_lo, the indexer will correctly skip
+    /// committing checkpoints less than the main pipeline's reader watermark.
+    #[tokio::test]
+    async fn test_tasked_pipelines_ignore_below_main_reader_lo() {
+        let cancel = CancellationToken::new();
+        let registry = Registry::new();
+        let store = MockStore::default();
+
+        // Mock the store as if we have a main pipeline with a committer watermark at `10` and a
+        // reader watermark at `7`.
+        let mut conn = store.connect().await.unwrap();
+        conn.set_committer_watermark(
+            MockCheckpointSequenceNumberHandler::NAME,
+            CommitterWatermark {
+                checkpoint_hi_inclusive: 10,
+                ..Default::default()
+            },
+        )
+        .await
+        .unwrap();
+        conn.set_reader_watermark(MockCheckpointSequenceNumberHandler::NAME, 7)
+            .await
+            .unwrap();
+
+        // Start a tasked indexer that will ingest from checkpoint 0. Checkpoints 0 through 6 should
+        // be ignored by the tasked indexer.
+        let indexer_args = IndexerArgs {
+            first_checkpoint: Some(0),
+            last_checkpoint: Some(15),
+            pipeline: vec![],
+            task: Some("task".to_string()),
+        };
+        let temp_dir = tempfile::tempdir().unwrap();
+        synthetic_ingestion::generate_ingestion(synthetic_ingestion::Config {
+            ingestion_dir: temp_dir.path().to_owned(),
+            starting_checkpoint: 0,
+            num_checkpoints: 16,
+            checkpoint_size: 2,
+        })
+        .await;
+
+        let client_args = ClientArgs {
+            local_ingestion_path: Some(temp_dir.path().to_owned()),
+            ..Default::default()
+        };
+
+        let ingestion_config = IngestionConfig::default();
+
+        let mut tasked_indexer = Indexer::new(
+            store.clone(),
+            indexer_args,
+            client_args,
+            ingestion_config,
+            None,
+            &registry,
+            cancel,
+        )
+        .await
+        .unwrap();
+
+        let _ = tasked_indexer
```

Comment:
I know this wasn't a pattern that was introduced by this PR, but it seems strange to me -- this is a test, so we should be able to use either `unwrap` or `unwrap_err` on any call, and not just eat the result.